### PR TITLE
metrics: collect disk usage from internal metrics

### PR DESCRIPTION
**Pull request by tomasmatus**
use `mount.total` and `mount.used` from internal metrics instead of parsing `df` output.

### CODE REVIEW

**Comment by jelly** [comment]
Fun times, you forget one important detail, the JS code filters mount points (loopback, flatpak, and? tmpfs?)

File: src/cockpit/samples.py [hunk 1]
```diff
@@ -398,7 +398,15 @@ def sample(self, samples: Samples) -> None:
                 if line[0] != '/':
                     continue
 
-                path = line.split()[1]
+                split_line = line.split()
+                # Skip read-only loop mounts
+                if '/loop' in split_line[0] and 'ro' in split_line[3].split(','):
```
Comment (jelly) [inline src/cockpit/samples.py:403]: Wouldn't it be nicer to properly decompose this

```python
mounts = open('/proc/mounts').read()
fs_spec, fs_file, fs_vfstype, fs_mntopts, *_rest  = mounts.splitlines()[0].split()
```

Not sure if one can remove `*_rest`, field definitions comes from https://man.archlinux.org/man/fstab.5

File: src/cockpit/samples.py [hunk 2]
```diff
@@ -398,7 +398,15 @@ def sample(self, samples: Samples) -> None:
...
+                    continue
+                # Hide flatpaks
+                if 'revokefs-fuse' in split_line[0] and 'flatpak' in split_line[1]:
+                    continue
```
Comment (jelly) [inline src/cockpit/samples.py:407]: What about:

```
tmpfs /run/credentials/systemd-journald.service tmpfs ro,nosuid,nodev,noexec,relatime,nosymfollow,size=1024k
```

Mount point exists, but we likely don't want it.

Comment (tomasmatus) [reply src/cockpit/samples.py:407]: Won't this get skipped by the:

```
# Only look at real devices
if line[0] != '/':
    continue
```

Comment (jelly) [reply src/cockpit/samples.py:407]: Yes :)

**Review by jelly — APPROVED** [review]
Nice cleanup!

**Pull request merged**